presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexSupplier.java

```diff
@@ -137,7 +144,16 @@ private static STRtree buildRTree(LongArrayList addresses, List<List<Block>> cha
                 partition = toIntExact(INTEGER.getLong(partitionBlock, blockPosition));
             }
 
-            rtree.insert(getJtsEnvelope(ogcGeometry, radius), new GeometryWithPosition(ogcGeometry, partition, position));
+            GeometryWithPosition geometryWithPosition = new GeometryWithPosition(ogcGeometry, partition, position);
+            rtree.insert(getJtsEnvelope(ogcGeometry, radius), geometryWithPosition);
+
+            estimatedMemorySizeInBytes += ENVELOPE_INSTANCE_SIZE + geometryWithPosition.getEstimatedMemorySizeInBytes();
```

Comment:
There is an array of references collecting the geometries inside the rtree. It's not 100% accurate, but can you add size of an object reference each time as well?

Reply:
Could you be more specific about which array ?  I only include leaf nodes into consideration, the total size will be off by 11%.

Reply:
Is there a reason no to be as precise as we can?

The rtree before you invoke build is wrapping what you provide in an ItemBoundable and stashing it in a big array which it then discards when the rtree is built.

This is a slightly more accurate accounting https://github.com/aweisberg/presto/blob/fix_spatial_index_accounting/presto-main/src/main/java/com/facebook/presto/operator/PagesSpatialIndexBuilder.java#L143

Reply:
11% more nodes are generated when insertion is done. I will add ItemBoundable. What is the 8-bytes array?

Reply:
What file/line in the rtree library generates more nodes on insertion? When I looked at the rtree code it just wraps it and stores it in an array.

Do you mean it uses more after it builds?

Reply:
https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/index/strtree/AbstractSTRtree.java#L243
https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/index/strtree/AbstractSTRtree.java#L109

Reply:
> Do you mean it uses more after it builds?
Yes,  when insertion is done, parent nodes will be created after final build

Reply:
createHigherLevels https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/index/strtree/AbstractSTRtree.java#L107